test/traversals/diffusion.jl

```diff
@@ -0,0 +1,181 @@
+
+    # Basic test. Watch connected vertices
+    result = @inferred(diffusion_rate(g,
+                                  1.0,
+                                  4,
+                                  watch=collect(1:5),
+                                  initial_infections=[2]
+                                  )
+             )
+    result
+    @test result == [1, 5, 5, 5]
+
+    # Watching unconnected vertices
+    result = @inferred(diffusion_rate(g,
+                                  1.0,
+                                  4,
+                                  watch=collect(6:10),
+                                  initial_infections=[2]
+                                  )
+               )
+    @test  result == [0, 0, 0, 0]
+
+    # Watch subset
+    result = @inferred(diffusion_rate(g,
+                                  1.0,
+                                  4,
+                                  watch=collect(1:2),
+                                  initial_infections=[2]
+                                  )
+             )
+    result
+    @test result == [1, 2, 2, 2]
+
+    result = diffusion_rate(g,
+                                  1.0,
+                                  4,
+                                  watch=collect(1:5),
+                                  initial_infections=[10]
+                                  )
+
+    @test result == [0, 0, 0, 0]
+
+    ######
+    # Check along path graph
+    ######
+
+    g2 = PathGraph(5)
+    result = @inferred(diffusion_rate(g2,
+                                  1.0,
+                                  4,
+                                  watch=collect(1:5),
+                                  initial_infections=[1]
+                                  )
+              )
+    @test  result == [1, 2, 3, 4]
+
+    result = @inferred(diffusion_rate(g2,
+                                  1.0,
+                                  4,
+                                  watch=collect(1:5),
+                                  initial_infections=[3]
+                                  )
+             )
+    @test result == [1, 3, 5, 5]
+
+    # Check normalize
+    g3 = PathGraph(30)
+    result = @inferred(diffusion_rate(g3,
+                                  1.0,
+                                  6,
+                                  initial_infections=[15],
+                                  normalize=false
+                                  )
+             )
+    result
+    @test result == [1, 3, 5, 7, 9, 11]
+
+
+    result = @inferred(diffusion_rate(g3,
+                                  2.0,
+                                  6,
+                                  initial_infections=[15],
+                                  normalize=true
+                                  )
+             )
+    @test result == [1, 3, 5, 7, 9, 11]
+
+    # Test probability accurate
+    means = Dict(0.2 => 0.8, 0.4 => 1.6)
+    stds = Dict(0.2 => 0.8, 0.4 => 0.98)
+    runs = 20
+    for p in [0.2, 0.4]
```

Comment:
I originally used `Distributions` to generate all these values in place, but didn't want to add dependency.